src/t_hash.c

```diff
@@ -406,22 +423,41 @@ static uint64_t listpackExGetMinExpire(robj *o) {
 }
 
 /* Walk over fields and delete the expired ones. */
-void listpackExExpire(robj *o, ExpireInfo *info) {
+void listpackExExpire(redisDb *db, robj *o, ExpireInfo *info) {
     serverAssert(o->encoding == OBJ_ENCODING_LISTPACK_EX);
-    uint64_t min = EB_EXPIRE_TIME_INVALID;
+    uint64_t expired = 0, min = EB_EXPIRE_TIME_INVALID;
+    unsigned char *ptr;
     listpackEx *lpt = o->ptr;
 
-    struct lpFingArgs r = {
-        .max_to_search = info->maxToExpire,
-        .expire_time = info->now
-    };
+    ptr = lpFirst(lpt->lp);
 
-    lpFindCb(lpt->lp, NULL, &r, cbFindInListpack, 0);
-    info->itemsExpired += r.expired;
+    while (ptr != NULL && (info->itemsExpired < info->maxToExpire)) {
+        long long val;
+        int64_t flen;
+        unsigned char intbuf[LP_INTBUF_SIZE], *fref;
+
+        fref = lpGet(ptr, &flen, intbuf);
```

Comment:
```c
int flen;
long long fval
unsigned char *fstr;

fstr = lpGetValue(ptr, &flen, &fval);
if (fstr)
    printf("%.*s", flen, fstr);
else
    printf("%lld", fval);
```

You may get the value like this. You don't need to provide `intbuf` as well, fyi.